dotCMS/src/main/java/com/dotcms/http/server/mock/MockHttpServerContext.java

```diff
@@ -0,0 +1,142 @@
+package com.dotcms.http.server.mock;
+
+
+import com.dotcms.http.CircuitBreakerUrlBuilder;
+import com.dotcms.repackage.com.sun.xml.ws.client.ResponseContext;
+import com.dotmarketing.util.UtilMethods;
+import com.liferay.util.StringPool;
+import com.sun.net.httpserver.Headers;
+import com.sun.net.httpserver.HttpExchange;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import org.apache.commons.io.IOUtils;
+
+public class MockHttpServerContext {
```

Comment:
Thjis class should live outside production code

Reply:
I think this is the only way to use it into Integration Test and Unit Test